### PR TITLE
Feat cleanup executor

### PR DESCRIPTION
**Pull request by PineappleIOnic**
## What does this PR do?

Cleanups up aspects of the executor to bring it more inline with appwrite itself

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
Yes, I have

### CODE REVIEW

**Review by christyjacob4 — APPROVED** [review]

File: app/executor.php [hunk 1]
```diff
@@ -1051,33 +1049,29 @@ function (string $functionId, Response $response, Database $dbForProject) use ($
...
+        } catch (Throwable $th) {
+            $orchestrationPool->put($orchestration);
+            throw $th;
         }
         $orchestrationPool->put($orchestration);
 
-        return $response->json(['success' => true]);
+        return $response
+            ->setStatusCode(Response::STATUS_CODE_OK)
+            ->json(['success' => true]);
```
Comment (christyjacob4) [inline app/executor.php:1074]:
```suggestion
        return $response
            ->setStatusCode(Response::STATUS_CODE_OK)
            ->send();
```

File: app/executor.php [hunk 2]
```diff
@@ -966,38 +965,37 @@ function (string $functionId, Response $response, Database $dbForProject) use ($
...
+                    $orchestration->remove('appwrite-function-' . $deployment['$id'], true);
+                    Console::info('Removed container for deployment ' . $deployment['$id']);
                 }
 
-                return $response->json(['success' => true]);
-            } catch (Exception $e) {
-                logError($e, "cleanupFunction");
+                return $response
+                    ->setStatusCode(Response::STATUS_CODE_OK)
+                    ->json(['success' => true]);
```
Comment (christyjacob4) [inline app/executor.php:992]:
```suggestion
                $response
                    ->setStatusCode(Response::STATUS_CODE_OK)
                    ->send();
```

File: app/executor.php [hunk 3]
```diff
@@ -966,38 +965,37 @@ function (string $functionId, Response $response, Database $dbForProject) use ($
 
                 // If amount is 0 then we simply return true
                 if (count($results) === 0) {
-                    return $response->json(['success' => true]);
+                    return $response
+                        ->setStatusCode(Response::STATUS_CODE_OK)
+                        ->json(['success' => true]);
```
Comment (christyjacob4) [inline app/executor.php:970]:
```suggestion
                    $response
                        ->setStatusCode(Response::STATUS_CODE_OK)
                        ->send();
```

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Please address the comments

File: app/executor.php [hunk 4]
```diff
@@ -915,46 +914,38 @@ function runBuildStage(string $buildId, string $projectID): Document
...
 // Cleanup Endpoints used internally by appwrite when a function or deployment gets deleted to also clean up their containers
 App::post('/v1/cleanup/function')
-    ->param('functionId', '', new UID())
+    ->param('functionId', '', new UID(), 'The FunctionID to cleanup')
     ->inject('response')
     ->inject('dbForProject')
     ->action(
         function (string $functionId, Response $response, Database $dbForProject) use ($orchestrationPool) {
+            /** @var Orchestration $orchestration */
+            $orchestration = $orchestrationPool->get();
```
[the remark this answers is not in the document]
Comment (christyjacob4) [reply app/executor.php]: Added it inside the `try`

File: app/executor.php [hunk 5]
```diff
@@ -1051,33 +1049,29 @@ function (string $functionId, Response $response, Database $dbForProject) use ($
...
+            // Remove any ongoing builds
+            if ($deployment->getAttribute('buildId')) {
+                $build = $dbForProject->getDocument('builds', $deployment->getAttribute('buildId'));
 
-                // Remove the container of the deployment
-                $orchestration->remove('appwrite-function-' . $deployment['$id'], true);
-                Console::info('Removed container for deployment ' . $deployment['$id']);
-            } catch (Exception $e) {
-                // Do nothing, we don't care that much if it fails
+                if ($build->getAttribute('status') == 'building') {
```
Comment (TorstenDittmann) [inline app/executor.php:1056]: Always to strict equal `===`

**Pull request merged**